src/utils/CookieAndLocalStorageUtils.ts

```diff
@@ -0,0 +1,21 @@
+import { Cookie } from './CookieUtils';
+import { SafeLocalStorage } from './LocalStorageUtils';
+
+// Copied from https://github.com/coveo/coveo.analytics.js/blob/master/src/storage.ts
+export class CookieAndLocalStorage implements Partial<Storage> {
+  private safeLocalStorage = new SafeLocalStorage();
+
+  getItem(key: string): string | null {
+    return this.safeLocalStorage.getItem(key) || Cookie.get(key);
+  }
+
+  removeItem(key: string): void {
+    this.safeLocalStorage.removeItem(key);
+    Cookie.erase(key);
+  }
+
+  setItem(key: string, data: string): void {
+    this.safeLocalStorage.setItem(key, data);
+    Cookie.set(key, data);
```

Comment:
If you don't set an expiration date on the cookie, the cookie will be session level and useless. coveo.analytics.js uses a 1-year expiry (which may be restricted depending on browser).